core/src/main/scala/org/apache/spark/deploy/worker/ui/LogPage.scala

```diff
@@ -60,16 +63,19 @@ private[ui] class LogPage(parent: WorkerWebUI) extends WebUIPage("logPage") with
     val appId = Option(request.getParameter("appId"))
     val executorId = Option(request.getParameter("executorId"))
     val driverId = Option(request.getParameter("driverId"))
+    val self = Option(request.getParameter("self"))
     val logType = request.getParameter("logType")
     val offset = Option(request.getParameter("offset")).map(_.toLong)
     val byteLength = Option(request.getParameter("byteLength")).map(_.toInt)
       .getOrElse(defaultBytes)
 
-    val (logDir, params, pageName) = (appId, executorId, driverId) match {
-      case (Some(a), Some(e), None) =>
+    val (logDir, params, pageName) = (appId, executorId, driverId, self) match {
+      case (Some(a), Some(e), None, None) =>
         (s"${workDir.getPath}/$a/$e/", s"appId=$a&executorId=$e", s"$a/$e")
-      case (None, None, Some(d)) =>
+      case (None, None, Some(d), None) =>
         (s"${workDir.getPath}/$d/", s"driverId=$d", d)
+      case (None, None, None, Some(_)) =>
+        (s"${sys.env.getOrElse("SPARK_LOG_DIR", workDir.getPath)}/", "self", "")
```

Comment:
Assign "" to `pageName` may produce a weird page title "...  log page for "?

Reply:
Thanks. let me check.

Reply:
This is fixed.